gamla/functional_generic.py

```diff
@@ -819,11 +819,13 @@ def groupby(
 def is_generator(iterable):
     return hasattr(iterable, "__iter__") and not hasattr(iterable, "__len__")
 
+
 def _transform_if_not_none(transform: Callable, value):
```

Comment:
1. move this to `data.py`
2. this private, so you won't be able to use it from outside... should be public.
3. add a docstring (see other functions), we make sure to document gamla properly
4. consider adding a short test